notify.go

```diff
@@ -0,0 +1,31 @@
+package autonat
+
+import (
+	inet "github.com/libp2p/go-libp2p-net"
+	peer "github.com/libp2p/go-libp2p-peer"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+var _ inet.Notifiee = (*AmbientAutoNAT)(nil)
+
+func (as *AmbientAutoNAT) Listen(net inet.Network, a ma.Multiaddr)      {}
+func (as *AmbientAutoNAT) ListenClose(net inet.Network, a ma.Multiaddr) {}
+func (as *AmbientAutoNAT) OpenedStream(net inet.Network, s inet.Stream) {}
+func (as *AmbientAutoNAT) ClosedStream(net inet.Network, s inet.Stream) {}
+
+func (as *AmbientAutoNAT) Connected(net inet.Network, c inet.Conn) {
```

Comment:
I think we can reduce the noise by simply checking on the protocols reported by identify.

Reply:
I changed it to look at the protocols reported by identify through the peerstore in https://github.com/libp2p/go-libp2p-autonat/pull/1/commits/46d352fc09280b21c7aea7dc02cf9da41f63dbc0
I don't quite like the delay for identify, but it seems to be necessary.

Reply:
>  I don't quite like the delay for identify, but it seems to be necessary.

Yeah... that annoys me to me to no end as well.

Reply:
`Identify` keeps the stream open after identification. If we made it close the stream, we could hook onto the `ClosedStream` event to know when identify was over more deterministically.

Reply:
Let's really not. I know it should work, but that's just horrible.

Reply:
It is, but that's all we have now ;-) Should we think about setting up some kind of "in-mem event bus" so that different layers of libp2p can emit and react to events? _Identify_  would then emit an "protocols:identify/1.0.0:complete" event when it finished, or a "...:error" one if it failed.

Reply:
Yeah... Ideally services would just hook into identify (or the peerstore? that doesn't seem right) and get called when we connect to a peer supporting protocol X.

Reply:
Made the delay configurable, with an initial value of 5 sec (per @magik6k's suggestion)